exporter/elasticsearchexporter/config.go

```diff
@@ -233,11 +201,17 @@ type DiscoverySettings struct {
 // FlushSettings defines settings for configuring the write buffer flushing
 // policy in the Elasticsearch exporter. The exporter sends a bulk request with
 // all events already serialized into the send-buffer.
+//
+// Deprecated: [v0.136.0] This config is now deprecated. Use `sending_queue::batch` instead.
+// Flush settings will be ignored if defined and will be dropped in future releases.
 type FlushSettings struct {
```

Comment:
[**For reviewers**] We have 3 options for `flush` and `num_consumers` settings:

1. Mark the settings deprecated and, in the interim, use them to define sending queue configurations
2. Remove the settings enitrely as breaking change even though they are not currently deprecated
3. Mark the settings deprecated and ignore them - what the PR does

I don't want to do `1` and I am conflicted between `2` and `3` but I don't have a strong opinion on this so happy to hear what othres think.

Reply:
(3) is acceptable but my first instinct would be (1) although I don't love like it either. (1) will follow a proper deprecation process which involves approximating equiv config. If we were to go with (3), it warrants a WARN log on startup in `handleDeprecatedConfig` etc

Reply:
> I prefer (1) -- deprecating and ignoring the setting in the same release isn't very user friendly.

Fair point, let me take a stab at implmenting parity with the configurations.

Reply:
Done in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/42767/commits/1db0f2b07338e13de23355e902ec09ad5268abac